src/ec2/face_detector.py

```
@@ -14,33 +15,50 @@ def __init__(self, output_dir="detected_faces", tolerance=0.6):
         os.makedirs(self.output_dir, exist_ok=True)
         self.tolerance = tolerance
         self.known_embeddings = []
+        self.models = ['buffalo_l', 'buffalo_m', 'buffalo_s', 'buffalo_sc', 'antelopev2']
+        self.current_model = self.models[0]
         try:
-            self.face_app = FaceAnalysis(name='buffalo_l', providers=['CPUExecutionProvider'], root="/tmp/.insightface")
+            self.face_app = FaceAnalysis(name=self.current_model, providers=['CPUExecutionProvider'], root="/tmp/.insightface")
             self.face_app.prepare(ctx_id=0, det_size=(640, 640))
         except Exception as e:
             raise RuntimeError(f"Failed to initialize InsightFace: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)